sdk/quoting/src/example/Example.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useCallback } from 'react'
+import './Example.css'
+import { ethers } from 'ethers'
+import { Token } from '@uniswap/sdk-core'
+import { CurrentConfig, Environment } from '../config'
+import { computePoolAddress } from '@uniswap/v3-sdk'
+import Quoter from '@uniswap/v3-periphery/artifacts/contracts/lens/Quoter.sol/Quoter.json'
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json'
+import { POOL_FACTORY_CONTRACT_ADDRESS, QUOTER_CONTRACT_ADDRESS } from '../libs/constants'
+import { getProvider, getWalletAddress, connectBrowserExtensionWallet } from '../libs/providers'
+import { toReadableAmount, fromReadableAmount } from '../libs/conversion'
+
+const getPoolConstants = async (): Promise<{ token0: string; token1: string; fee: number }> => {
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error('Provider not found')
+  }
+  const currentPoolAddress = computePoolAddress({
+    factoryAddress: POOL_FACTORY_CONTRACT_ADDRESS,
+    tokenA: CurrentConfig.currencies.in as Token,
+    tokenB: CurrentConfig.currencies.out as Token,
+    fee: CurrentConfig.currencies.fee,
+  })
+
+  const poolContract = new ethers.Contract(currentPoolAddress, IUniswapV3PoolABI.abi, provider)
+  const [token0, token1, fee] = await Promise.all([poolContract.token0(), poolContract.token1(), poolContract.fee()])
+
+  return {
+    token0,
+    token1,
+    fee,
+  }
+}
+
+const quote = async (setOutputAmount: (outputAmount: number) => void) => {
```

Comment:
Can we return the value instead? The goal for this function should be to not use React at all. It could even take in the tokens if we wanted to show it fully functional, though I think that just adds mess so I'm good with doing only the return and then setting it in a `useCallback`